pym/bob/input.py

```diff
@@ -1983,12 +1985,17 @@ def prepare(self, pathFormatter, inputEnv, sandboxEnabled, inputStates, inputSan
 
             p2 = thisDeps.setdefault(dep.recipe, p)
             if p2 is p:
-                # new pacakage
+                # new package
                 directPackages.append(p)
                 # add provided dependencies at the end
                 if dep.useDeps: indirectPackages.extend(p._getProvidedDeps())
                 # result is only picked once
-                if dep.useBuildResult: results.append(p)
+                if dep.useBuildResult:
+                    p2 = inResults.setdefault(dep.recipe, p)
+                    if p2 is p:
+                        results.append(p)
```

Comment:
This check is not needed. Each package (name) can only show up once here because of the `thisDeps.setdefault()` above.